server/app/services/openapi/AbstractOpenApiSchemaGenerator.java

```diff
@@ -28,4 +47,159 @@ protected ImmutableList<Scheme> getSchemes() {
   protected String getHostName() {
     return openApiSchemaSettings.baseUrl().replace("https://", "").replace("http://", "");
   }
+
+  protected static Property getPropertyFromType(
+      DefinitionType definitionType,
+      Optional<Format> swaggerFormat,
+      DefinitionType arrayItemDefinitionType) {
+
+    if (definitionType == DefinitionType.ARRAY) {
+      return new ArrayProperty()
+          .items(
+              arrayItemDefinitionType == DefinitionType.STRING
+                  ? new StringProperty()
+                  : new ObjectProperty());
+    }
+
+    var property = new UntypedProperty();
+
+    property.setType(definitionType.toString());
+
+    if (swaggerFormat.isPresent()) {
+      property.setFormat(swaggerFormat.get().toString());
+    }
+
+    return property;
+  }
+
+  /** Get the list of scalars for a question sorted alphabetically */
+  protected static ImmutableList<Scalar> getScalarsSortedByName(
+      QuestionDefinition questionDefinition)
+      throws InvalidQuestionTypeException, UnsupportedQuestionTypeException {
+    return Scalar.getScalars(questionDefinition.getQuestionType()).stream()
+        .filter(scalar -> !excludeScalarFromSchemaOutput(scalar))
+        .map(AbstractOpenApiSchemaGenerator::remapScalar)
+        .sorted(Comparator.comparing(Enum::name))
+        .collect(ImmutableList.toImmutableList());
+  }
```

Comment:
So this is going to result in any new scalars to a question automatically being added to the API, right?

I moved us away from that in the JSON/CSV (we now manually check each scalar we want to output) in part because of the mistake requiring the comment on line 99, but I'm ok with it here since we're also mapping them to types that aren't all strings.

what do you think?

Reply:
Ok, so going to leave this for now and look into adding some tests to validate the output of an api call with the schema if the openapi library supports it.